apps/wallet/src/ui/app/experimentation/feature-gating.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { GrowthBook } from '@growthbook/growthbook';
+
+const DEFAULT_DEV_API_KEY = 'key_dev_dc2872e15e0c5f95';
+
+export default class FeatureGating {
+    #growthBook: GrowthBook;
+
+    constructor() {
+        // Create a GrowthBook context
+        this.#growthBook = new GrowthBook();
+    }
+
+    public async init() {
+        const apiKey = process.env.GROWTH_BOOK_API_KEY ?? DEFAULT_DEV_API_KEY;
+        // Load feature definitions
+        await fetch(`https://cdn.growthbook.io/api/features/${apiKey}`)
```

Comment:
I'm fine with just always fetching this for now, but I think that we should persist these into a store so that we can re-use previous values in the event the API is down or something.